### PR TITLE
python3Packages.gradient: build fix

### PR DESCRIPTION
**Pull request by urbas**
###### Motivation for this change

The main motivation was to fix [this  failing build](https://hydra.nixos.org/build/134967105/log).

I think these are the minimal changes to fix the build. It's a bit concerning that so many new packages and so much renaming had to be done to fix this.

@freezeboy: Shall we just remove the old `gradient-sdk` and `paperspace` packages instead of adding all of these new packages? No other package depends on them, so removal should be easy.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by freezeboy** [comment]
I am ok with removing the old packages that were broken iirc btw.

I can't remember why it took so long and as i am on travel i can't check right now.

**Comment by urbas** [comment]
> I am ok with removing the old packages that were broken iirc btw.

Awesome, here's the pull request for the removal: #110576 

> I can't remember why it took so long and as i am on travel i can't check right now.

No worries, I'll leave this pull request open until we decide whether to remove or keep these packages.

File: pkgs/development/python-modules/log-symbols/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+, lib }:
+
+buildPythonPackage rec {
+  pname = "log-symbols";
+  version = "0.0.14";
+  disabled = isPy27;
+
+  src = fetchPypi {
+    # the package is named log-symbols but PyPI files are named log_symbols
+    pname = "log_symbols";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/log-symbols/default.nix:15]: I think the pname should be with _ but in top level python it should be with -. @jonringer

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110565` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.gradient</li>
<li>python37Packages.gradient-utils</li>
<li>python37Packages.halo</li>
<li>python37Packages.log-symbols</li>
<li>python37Packages.spinners</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.halo</li>
<li>python38Packages.log-symbols</li>
<li>python38Packages.spinners</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.halo</li>
<li>python39Packages.log-symbols</li>
<li>python39Packages.spinners</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

File: pkgs/development/python-modules/log-symbols/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/log-symbols/default.nix:15]: this is correct. Some packages were added before PEP 503 was ratified, as long as the attr in python-packages.nix adheres to PEP503, then it's good.

**Comment by urbas** [comment]
@SuperSandro2000 @jonringer: before I address the comments:

Shall we rather remove the packages (by merging #110576) instead of adding all these new packages (by merging this PR)?

**Comment by SuperSandro2000** [comment]
> Shall we rather remove the packages (by merging #110576) instead of adding all these new packages (by merging this PR)?

Adding 3 packages is in the grand scheme of nixpkgs nothing to worry about. If the package is useful we can keep it. Is there any other reason to remove it?

**Comment by urbas** [comment]
Great, no reason (I was just worried about how this simple fix blew up a bit).

Anyway, I'll go ahead with this PR and close the other one. :+1:

**Comment by jonringer** [comment]
> Is there any other reason to remove it?

every package is a small maintenance burden. I would just leave a comment near pname stating that it's no longer strictly needed, and can be safely removed.

If it can be maintained through nixpkgs-update (the bot), then I'm fine with keeping it around.

File: pkgs/development/python-modules/log-symbols/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/development/python-modules/log-symbols/default.nix:15]: also, this gets embedded into the url used to fetch the package, so it has to match the pypi listing.

**Comment by urbas** [comment]
@SuperSandro2000: adding tests will be a significant undertaking. I have refrained from adding them to keep the PR small. Let me know if I should still go ahead and add tests.

Other than that the PR is ready to go.

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000: adding tests will be a significant undertaking. I have refrained from adding them to keep the PR small. Let me know if I should still go ahead and add tests.
> 
> Other than that the PR is ready to go.

It would be nice to have more testing but it is not a hard requirement.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110565` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.gradient</li>
<li>python37Packages.gradient-utils</li>
<li>python37Packages.halo</li>
<li>python37Packages.log-symbols</li>
<li>python37Packages.spinners</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.halo</li>
<li>python38Packages.log-symbols</li>
<li>python38Packages.spinners</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.halo</li>
<li>python39Packages.log-symbols</li>
<li>python39Packages.spinners</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110565` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.gradient</li>
<li>python37Packages.gradient-utils</li>
<li>python37Packages.halo</li>
<li>python37Packages.log-symbols</li>
<li>python37Packages.spinners</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.halo</li>
<li>python38Packages.log-symbols</li>
<li>python38Packages.spinners</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.halo</li>
<li>python39Packages.log-symbols</li>
<li>python39Packages.spinners</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.gradient-utils:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add `'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by urbas** [comment]
@SuperSandro2000: I added tests where I could (gradient-utils) and documented why I couldn't add tests to the rest.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110565` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
13 packages built:
</summary>
<ul>
<li>python37Packages.gradient</li>
<li>python37Packages.gradient-utils</li>
<li>python37Packages.halo</li>
<li>python37Packages.log-symbols</li>
<li>python37Packages.spinners</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.halo</li>
<li>python38Packages.log-symbols</li>
<li>python38Packages.spinners</li>
<li>python39Packages.halo</li>
<li>python39Packages.log-symbols</li>
<li>python39Packages.spinners</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110565` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
15 packages built:
</summary>
<ul>
<li>python37Packages.gradient</li>
<li>python37Packages.gradient-utils</li>
<li>python37Packages.halo</li>
<li>python37Packages.log-symbols</li>
<li>python37Packages.spinners</li>
<li>python38Packages.gradient</li>
<li>python38Packages.gradient-utils</li>
<li>python38Packages.halo</li>
<li>python38Packages.log-symbols</li>
<li>python38Packages.spinners</li>
<li>python39Packages.gradient</li>
<li>python39Packages.gradient-utils</li>
<li>python39Packages.halo</li>
<li>python39Packages.log-symbols</li>
<li>python39Packages.spinners</li>
</ul>
</details>
